Documentation and building steps aren't working

- [x] fix `doc`
- [x] fix `package`
- [x] fix `make`

----

On the current version of the code, on the `main` branch. There are some long, intimidating error messages, but I'm pretty sure most of these are just broken paths.

1. For `npm run doc`:

```
> tokyo-drift@1.0.0 doc /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift
> documentation build src/** -f html -o docs

Error: Cannot find module '../index.css' from '/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/src/renderer'
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/browser-resolve/node_modules/resolve/lib/async.js:55:21
    at load (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/browser-resolve/node_modules/resolve/lib/async.js:69:43)
    at onex (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/browser-resolve/node_modules/resolve/lib/async.js:92:31)
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/browser-resolve/node_modules/resolve/lib/async.js:22:47
    at FSReqCallback.oncomplete (fs.js:176:21)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tokyo-drift@1.0.0 doc: `documentation build src/** -f html -o docs`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tokyo-drift@1.0.0 doc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

2. For `npm run package`:

```
> tokyo-drift@1.0.0 package /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift
> electron-forge package

✔ Checking your system
✖ Compiling Main Process Code

An unhandled error has occurred inside Forge:
Compilation errors in the main process: Hash: 4ed7c5125be4905daf96
Version: webpack 4.44.2
Time: 509ms
Built at: 12/09/2020 7:20:41 PM
                   Asset      Size  Chunks         Chunk Names
                index.js  8.97 KiB       0         app
            index.js.map  34.5 KiB       0  [dev]  app
native_modules/home.html  4.02 KiB                 
Entrypoint app = index.js index.js.map
 [0] external "electron" 42 bytes {0} [built]
 [1] external "path" 42 bytes {0} [built]
 [2] ./node_modules/debug/src/debug.js 4.29 KiB {0} [built]
 [3] ./src/main.js 1.87 KiB {0} [built]
 [4] ./node_modules/electron-squirrel-startup/index.js 1 KiB {0} [built]
 [5] external "child_process" 42 bytes {0} [built]
 [6] ./node_modules/debug/src/index.js 263 bytes {0} [built]
 [7] ./node_modules/debug/src/browser.js 4.62 KiB {0} [built]
 [8] ./node_modules/ms/index.js 2.7 KiB {0} [built]
 [9] ./node_modules/debug/src/node.js 5.87 KiB {0} [built]
[10] external "tty" 42 bytes {0} [built]
[11] external "util" 42 bytes {0} [built]
[12] external "fs" 42 bytes {0} [built]
[13] external "net" 42 bytes {0} [built]
[14] external "url" 42 bytes {0} [built]

ERROR in Entry module not found: Error: Can't resolve '/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/src/timer.js' in '/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift'
Error: Compilation errors in the main process: Hash: 4ed7c5125be4905daf96
Version: webpack 4.44.2
Time: 509ms
Built at: 12/09/2020 7:20:41 PM
                   Asset      Size  Chunks         Chunk Names
                index.js  8.97 KiB       0         app
            index.js.map  34.5 KiB       0  [dev]  app
native_modules/home.html  4.02 KiB                 
Entrypoint app = index.js index.js.map
 [0] external "electron" 42 bytes {0} [built]
 [1] external "path" 42 bytes {0} [built]
 [2] ./node_modules/debug/src/debug.js 4.29 KiB {0} [built]
 [3] ./src/main.js 1.87 KiB {0} [built]
 [4] ./node_modules/electron-squirrel-startup/index.js 1 KiB {0} [built]
 [5] external "child_process" 42 bytes {0} [built]
 [6] ./node_modules/debug/src/index.js 263 bytes {0} [built]
 [7] ./node_modules/debug/src/browser.js 4.62 KiB {0} [built]
 [8] ./node_modules/ms/index.js 2.7 KiB {0} [built]
 [9] ./node_modules/debug/src/node.js 5.87 KiB {0} [built]
[10] external "tty" 42 bytes {0} [built]
[11] external "util" 42 bytes {0} [built]
[12] external "fs" 42 bytes {0} [built]
[13] external "net" 42 bytes {0} [built]
[14] external "url" 42 bytes {0} [built]

ERROR in Entry module not found: Error: Can't resolve '/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/src/timer.js' in '/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift'
    at cb (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/@electron-forge/plugin-webpack/src/WebpackPlugin.ts:250:31)
    at finalCallback (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/webpack/lib/Compiler.js:257:39)
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/webpack/lib/Compiler.js:273:13
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/Hook.js:154:20)
    at onCompiled (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/webpack/lib/Compiler.js:271:21)
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/webpack/lib/Compiler.js:681:15
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/Hook.js:154:20)
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/webpack/lib/Compiler.js:678:31
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/Hook.js:154:20)
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/webpack/lib/Compilation.js:1423:35
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/tapable/lib/Hook.js:154:20)
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/webpack/lib/Compilation.js:1414:32
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tokyo-drift@1.0.0 package: `electron-forge package`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tokyo-drift@1.0.0 package script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

3. For `npm run make`:

```
> tokyo-drift@1.0.0 make /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift
> electron-forge make

✔ Checking your system
✔ Resolving Forge Config

An unhandled rejection has occurred inside Forge:
Error: Cannot make for rpm, the following external binaries need to be installed: rpmbuild
    at MakerRpm.ensureExternalBinariesExist (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/@electron-forge/maker-base/src/Maker.ts:147:13)
    at _default (/home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/@electron-forge/core/src/api/make.ts:148:11)
    at /home/marcus/Dropbox/School/UCSD/Year3/Fall2020/CSE210/project/fedarko-tokyo-drift/tokyo-drift/node_modules/@electron-forge/cli/src/electron-forge-make.ts:44:5

Electron Forge was terminated. Location:
{}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! tokyo-drift@1.0.0 make: `electron-forge make`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the tokyo-drift@1.0.0 make script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

--- a/README.md
+++ b/README.md
@@ -31,20 +31,12 @@ npm install
 
 See the repository's `package.json` file.
 
-### Start development
+### Start the application
 
 You can start the app with:
 
 ```
 npm start
-```
-
-The app has hot-reloading for any JS or HTML change. For any change to the electron code, just type `rs` in the electron console.
-
-For debugging, you can add the following statement to your electron window:
-
-```
-win.webContents.openDevTools()
 ```
 
 ### Linting and stylechecking
@@ -79,28 +71,58 @@ npm run test:unit
 
 ### Generate the documentation
 
+Documentation can be auto-generated using
+[documentation.js](https://documentation.js.org/), a lovely library that does
+this with aplomb.
+
+The following command will generate pretty, easy-to-navigate documentation for
+all of the various functions/classes included within the codebase:
+
 ```
 npm run doc
 ```
 
-All the documentation should be available after that in the `docs/` directory in the HTML file format.
+All the documentation should be available after that in the `docs/` directory
+in the HTML file format.
+
+We note that it's currently the job of the individual developer to generate
+documentation on their own system; in the future, it should be totally possible
+to do something like generate the documentation in the CI pipeline and publish
+it to a GitHub pages site.
 
 ### Build the application
 
-**NOTE: This is currently not working. We're hoping to fix this soon.**
+We do this using [Electron Forge](https://www.electronforge.io/).
 
-For the moment, this is only available for Linux distributions.
+For the moment, this is only supported for Linux distributions, but (since
+Electron Forge is pretty comprehensive) this should be extendable relatively
+easily to work with other operating systems.
 
-You can either make a package containing all the files of the app:
+There are two main "build" commands we have set up right now:
+
+#### Option 1: Build a package containing all the files of the app
 
 ```
 npm run package
 ```
 
-Or you can gather everything into an executable:
+#### Option 2: Gather everything into an executable
 
 ```
 npm run make
 ```
 
-All the result will be available in the `out/` directory.
+#### Accessing the output of building
+
+All the results will be available in the `out/` directory.
+
+#### Troubleshooting building
+
+If you get an error from `electron-forge` in the above steps (something along
+the lines of "Cannot make for rpm, the following external binaries need to be
+installed"), then you need to make sure that certain packages are installed on
+your system.
+
+For one of the developers using Ubuntu, installing `rpm` using
+`sudo apt install rpm` fixed this problem
+([shoutouts to Stack Overflow](https://stackoverflow.com/a/59571436)).

